File: solution/0100-0199/0165.Compare Version Numbers/Solution.cs
using System;
using System.Linq;
using System.Collections.Generic;

public class Solution {
    public int CompareVersion(string version1, string version2) {
        var parts1 = version1.Split('.').Select(int.Parse).ToList();
        var parts2 = version2.Split('.').Select(int.Parse).ToList();
        var minLen = Math.Min(parts1.Count, parts2.Count);
        for (var i = 0; i < minLen; ++i)
        {
            if (parts1[i] < parts2[i])
            {
                return -1;
            }
            else if (parts1[i] > parts2[i])
            {
                return 1;
            }
        }
        if (parts1.Count > parts2.Count && parts1.Skip(minLen).Any(v => v != 0))
        {
            return 1;
        }
        if (parts2.Count > parts1.Count && parts2.Skip(minLen).Any(v => v != 0))
        {
            return -1;
        }
        return 0;
    }
}

File: solution/0100-0199/0165.Compare Version Numbers/Solution.go
func compareVersion(version1 string, version2 string) int {
	m, n := len(version1), len(version2)
	for i, j := 0, 0; i < m || j < n; i, j = i+1, j+1 {
		var a, b int
		for i < m && version1[i] != '.' {
			a = a*10 + int(version1[i]-'0')
			i++
		}
		for j < n && version2[j] != '.' {
			b = b*10 + int(version2[j]-'0')
			j++
		}
		if a < b {
			return -1
		}
		if a > b {
			return 1
		}
	}
	return 0
}